src/transformers/tokenization_utils.py

```diff
@@ -956,3 +956,19 @@ def _decode(
             return clean_text
         else:
             return text
+
+
+class SentencePieceStringConversionMixin:
```

Comment:
I completely understand why you proposed to create a `mixin` class for this method common to many tokenizers. It seems that for the moment the philosophy of the transformers library is rather to duplicate the code than to make inheritances.

I would love to know the opinion of the `transformers` maintainers on this question. @LysandreJik , @sgugger , @patrickvonplaten  or @patil-suraj what you think about this design?

Reply:
For a short method like this, I would advise to put it in every tokenization file. Users don't like to have to read superclasses code and like that all the code is available in one place, so it's a core principle of our philosophy to have all code in one single file for the obecets of the library.

Reply:
No problem, I'm all in on that philosophy, especially since there are, I think, 3 special cases where I had to almost duplicate the code anyway.

For context, I decided on the mixin approach to try and mimic the thought process behind `GenerationMixin` and other such `*Mixin` classes I stumbled upon throughout the codebase exploration.

Reply:
Great :smile:, thank you so much!